client/src/components/Forms/UserForm/UserForm.tsx

```diff
@@ -0,0 +1,85 @@
+import { Buttons } from '@components/Button/Button';
+import FormField from '@components/FormField/FormField';
+import { MultiSelect } from '@components/MultiSelect/MultiSelect';
+import {
+    GetSelectDataType,
+    IOption,
+    MultiValueType,
+    OnChangeMultiValueType
+} from '@interfaces/types';
+import { IUser } from '@redux/types/users';
+import { UserFormFields } from '@utils/constants/UserFormFields';
+import { parceOption } from '@utils/functions/userFormData/parseOption';
+import { FC } from 'react';
+import { Form } from 'react-final-form';
+import { ButtonContainer, FieldCustom, FormContainer, SelectFieldCustom } from './styled';
+
+interface IProps {
+    currentUser?: any;
+    arrGroupNames?: string[];
+    getSelectData?: GetSelectDataType;
+    onChangeMultiValue: OnChangeMultiValueType;
+    selectGroupArr?: MultiValueType;
+    onSubmitForm: (value: IUser) => void;
+}
+
+export const UserForm: FC<IProps> = ({
+    currentUser,
+    getSelectData,
+    arrGroupNames,
+    selectGroupArr,
+    onChangeMultiValue,
+    onSubmitForm
+}) => {
+    const options: IOption[] = parceOption(arrGroupNames);
```

Comment:
how many items parsed here? mb it must be inside use memo

Reply:
not more 10 items, and it always rerender